### PR TITLE
[Flax] Add FlaxBlenderbotSmall

### LINKED ISSUE #14188: [Flax] Add Flax implementation of `BlenderbotSmallModel`
# 🚀 Feature request

Add Flax implementation of `BlenderbotSmall` models.

## Motivation

Improve the level of Flax/Jax support.

## Your contribution

I'll be happy to do it once the problems in #13633 will be resolved and the PR will be merged.

@patrickvonplaten

### PR DESCRIPTION
**Pull request by stancld**
# What does this PR do?

This PR adds `FlaxBlenderbotSmall` models.

**TODO:** Add model checkpoints for flax on the Hub.

Fixes #14188 

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@patrickvonplaten @patil-suraj

### CODE REVIEW

File: src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py [hunk 1]
```diff
@@ -0,0 +1,1525 @@
+# coding=utf-8
+# Copyright 2021 The Facebook, Inc. and The HuggingFace Inc. team and The HuggingFace Inc. team. All rights reserved.
```
Comment (patil-suraj) [inline src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py:2]:
```suggestion
# Copyright 2021 The Facebook, Inc. and The HuggingFace Inc. team. All rights reserved.
```

File: src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py [hunk 2]
```diff
@@ -0,0 +1,1525 @@
...
+                embedding_init=jax.nn.initializers.normal(self.config.init_std),
+            )
+
+        self.embed_positions = nn.Embed(
+            self.config.max_position_embeddings,
+            embed_dim,
+            embedding_init=jax.nn.initializers.normal(self.config.init_std),
+        )
+        self.layers = FlaxBlenderbotSmallEncoderLayerCollection(self.config, self.dtype)
+        self.layernorm_embedding = nn.LayerNorm(dtype=self.dtype)
```
Comment (patil-suraj) [inline src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py:707]:
```suggestion
        self.layernorm_embedding = nn.LayerNorm(dtype=self.dtype, epsilon=1e-05)
```

File: src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py [hunk 3]
```diff
@@ -0,0 +1,1525 @@
...
+            )
+
+        self.embed_positions = nn.Embed(
+            self.config.max_position_embeddings,
+            embed_dim,
+            embedding_init=jax.nn.initializers.normal(self.config.init_std),
+        )
+
+        self.layers = FlaxBlenderbotSmallDecoderLayerCollection(self.config, self.dtype)
+        self.layernorm_embedding = nn.LayerNorm(dtype=self.dtype)
```
Comment (patil-suraj) [inline src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py:776]:
```suggestion
        self.layernorm_embedding = nn.LayerNorm(dtype=self.dtype, epsilon=1e-05)
```

**Review by patil-suraj — APPROVED** [review]
LGTM! Thanks a lot for adding this model.

**Comment by patrickvonplaten** [comment]
Good to merge for me! Sorry for being so late and causing the merge conflict now @stancld :-/ Could you fix it quickly? Then I think we're good to merge this PR

**Review by patrickvonplaten — APPROVED** [review]

**Comment by stancld** [comment]
@patrickvonplaten No worries, it's okay :] Merge conflict should be resolved.

**Comment by patil-suraj** [comment]
Thanks, @stancld! Will push the checkpoints shortly and merge :)

File: tests/test_modeling_flax_blenderbot_small.py [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+                        outputs = decode_jitted(**prepared_inputs_dict).to_tuple()
+
+                self.assertEqual(len(outputs), len(jitted_outputs))
+                for jitted_output, output in zip(jitted_outputs, outputs):
+                    self.assertEqual(jitted_output.shape, output.shape)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_class_name in self.all_model_classes:
+            model = model_class_name.from_pretrained("facebook/blenderbot_small-90M", from_pt=True)
```
Comment (patil-suraj) [inline tests/test_modeling_flax_blenderbot_small.py:394]:
```suggestion
            model = model_class_name.from_pretrained("facebook/blenderbot_small-90M")
```

**Pull request merged**